[TrilinosApplication][SeveralApplications] Giving support for new TrilinosBlockBuilderAndSolvers in solvers

loumalouomega opened the pull request:
**📝 Description**

 Giving support for new `TrilinosBlockBuilderAndSolver` in solvers. In addition to that, support for all serial B&S has been added when missing in the other applications.

This required to add new default settings, and the corresponding validation as well.

**🆕 Changelog**

- [Update the structural solver](https://github.com/KratosMultiphysics/Kratos/commit/587f3174e1069ca751cb741960b73333df325cde)
- [Update convention diffusion solver](https://github.com/KratosMultiphysics/Kratos/commit/60e12d8d3ad395ee74af48fb394da93e9b7237d5)
- [Update fluid scripts](https://github.com/KratosMultiphysics/Kratos/commit/1f33320d6cf99c0dbe18a56844c17f7236d29c2c)

loumalouomega committed:
Update the structural solver

loumalouomega committed:
Missing clean up

loumalouomega committed:
Update convention diffusion solver

loumalouomega committed:
Missing warning

loumalouomega committed:
Missing validation of subparameters

loumalouomega committed:
Update fluid scripts

loumalouomega committed:
Missing validation

loumalouomega commented on applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_vmsmonolithic.py:
```diff
@@ -131,12 +131,19 @@ def _CreateBuilderAndSolver(self):
                 guess_row_size,
                 trilinos_linear_solver,
-                KratosFluid.PATCH_INDEX)
+                KratosCFD.PATCH_INDEX)
```
This was wrong before, maybe I can cherry-pick the fix if needed

loumalouomega committed:
Missing check

philbucher commented on applications/StructuralMechanicsApplication/python_scripts/trilinos_structural_mechanics_solver.py:
```diff
@@ -80,26 +79,23 @@ def _CreateLinearSolver(self):
-                                                                                         guess_row_size,
-                                                                                         linear_solver)
+            if self.settings["multi_point_constraints_used"].GetBool():
+                raise Exception("MPCs not yet implemented in MPI")
```
you added this in the meantime no?

rubenzorrilla commented on applications/ConvectionDiffusionApplication/python_scripts/convection_diffusion_solver.py:
```diff
@@ -128,9 +128,14 @@ def GetDefaultParameters(cls):
-            "block_builder": true,
+            "builder_and_solver_settings" : {
+                "use_block_builder" : true,
+                "use_lagrange_BS"   : false,
```
```suggestion
                "use_lagrange_builder_and_solver"   : false,
```
to follow the convention. Indeed, we are already inside the B&S settings so I'd put
```suggestion
                "use_lagrange_multipliers"   : false,
```
to avoid being redundant.

loumalouomega replied:
Then we should change it in the StructuralApplication

RiccardoRossi replied:
what does it mean "use_lagrange_bs"?

loumalouomega replied:
use_lagrange_builder_and_solver. I t is the key currently used in the StructuralMechanicsApplication

RiccardoRossi replied:
i was referring to "what does that BS do?"

loumalouomega replied:
It was introduced several years ago, it solves using LM for constraints instead of the Felippa algoritihm.

rubenzorrilla reviewed (commented):
Just saying that thinking in the long run, introducing the B&S settings in the solver ones is not the best solution. Instad, I'd only set the B&S name (i.e. registro points) and take these from the registry. Of course this is out of this PR context.

rubenzorrilla reviewed (changes requested):
Why adding the Lagrange BS to all the CFD and Heat solvers which don't actually use it?

sunethwarna committed:
initial commit

sunethwarna committed:
minor hierarchy change

sunethwarna committed:
remove from top

sunethwarna committed:
add order to Kratos menu

ncrescenzio committed:
link to kratos workshop event website

ncrescenzio committed:
kratos workshop html files

ncrescenzio committed:
remove dependence on yml files

ncrescenzio committed:
update speakers

ncrescenzio committed:
update venue

ncrescenzio committed:
update organizers

ncrescenzio committed:
minor update

ncrescenzio committed:
update schedule

ncrescenzio committed:
update schedule

ncrescenzio committed:
minor

ncrescenzio committed:
update schedule

ncrescenzio committed:
update venue

ncrescenzio committed:
update contact informations

ncrescenzio committed:
`Register Here` instead of `Register Now`

ncrescenzio committed:
typo

ncrescenzio committed:
update links

ncrescenzio committed:
update schedule

ncrescenzio committed:
update schedule

loumalouomega committed:
Merge master and fix conflict

loumalouomega committed:
Missing configuration

ncrescenzio committed:
removed material section

rfaasse committed:
Fix iterator to pointer issue in ublas_space.h (#12761)

loumalouomega committed:
Refactor FluidSolver settings validation and default assignment

This commit refactors the validation and default assignment of the `builder_and_solver_settings` parameter in the `FluidSolver` class. Previously, the code checked if the parameter existed and then validated and assigned defaults. Now, the code first checks if the parameter is missing and adds an empty value if necessary. Then, it proceeds to validate and assign defaults to the parameter.

The changes were made in the `applications/FluidDynamicsApplication/python_scripts/fluid_solver.py` file.

Co-authored-by: [Author Name]

loumalouomega committed:
Merge branch 'master' into trilinos/update-Trilinos-b&s-scripts-definition
